lib/connect/ws.js

```diff
@@ -41,12 +56,18 @@ function buildBuilder (client, opts) {
   }
 
   /*delete some mqtt overhead*/
-  mqttProperties.forEach(function (prop) {
-    if (opts.hasOwnProperty(prop)) {
-      delete opts[prop];
+  Object.keys(opts).forEach(function (prop) {
+    var propInKeys = false;
+    mqttProperties.concat(urlKeys).forEach(function(key){
+        if(prop === key){
+          propInKeys = true;
+        }
+    });
+    if (!propInKeys){
+      wsOpt[prop]=opts[prop];
     }
   })
```

Comment:
Yes, would be better. The Problem that I can't set the property e.g. protocolVersion for websockets is still not solved because we can't decide if the mqtt protocolVersion or the websocket protocolVersion is ment. 
Have a look at #512. There we can do stuff like that.